include/jemalloc/internal/hpa_opts.h

```diff
@@ -7,8 +7,53 @@
 /*
  * This file is morally part of hpa.h, but is split out for header-ordering
  * reasons.
+ *
+ * All of these hpa_shard_opts below are experimental. We are exploring more
+ * efficient packing, hugifying, and purging approaches to make efficient
+ * trade-offs between CPU, memory, latency, and usability. This means all of
+ * them are at the risk of being deprecated and corresponding configurations
+ * should be updated once the final version settles.
  */
 
+/*
+ * This enum controls when jemalloc hugifies/dehugifies pages.  Each mode may be
+ * more suitable depending on deployment environments.
+ *
+ * hpa_hugify_style_none
+ * This style only makes sense when deploying on systems where THP are enabled
+ * in 'always' mode.  With this style, you most likely want to have no purging
+ * at all (dirty_mult=-1) or purge_threshold=HUGEPAGE (2097152 for 2Mb page),
+ * although other thresholds may work well depending on kernel settings of your
+ * deployment targets.
+ *
+ * hpa_hugify_style_eager
+ * This style will set 'hg' vm_flag immediately after memory is mapped, and in
+ * case when THP is enabled in 'madvise' mode give you ability to start with
```

Comment:
The setting and clearing of flags is indirect but we directly provide hugepage advice to the memory region.  Maybe we could say that?  Something like, "Gives hugepage advice, if needed, to anonymous memory immediately after it is allocated, so huge pages can be allocated at page-fault time."